File: GenericTesting/VB_MVVM_Tester/Business/PlotPoints.vb
Public Class PlotPoints
  Inherits BaseViewModel

  'Private _x As PlotPoint
  Public ReadOnly Property X As PlotPoint
  Public ReadOnly Property Y As PlotPoint

  Public Sub New(x As PlotPoint, y As PlotPoint)
    Me.X = x
    Me.Y = y
  End Sub

  Public ReadOnly Property XAsDouble As Double
    Get
      Dim val As Double = 0

      Select Case True
        Case X.PointType Is GetType(Integer)
          val = CDbl(DirectCast(X, PlotPoint(Of Integer)).Point)
        Case X.PointType Is GetType(Double)
          val = CDbl(DirectCast(X, PlotPoint(Of Double)).Point)
        Case X.PointType Is GetType(Decimal)
          val = CDbl(DirectCast(X, PlotPoint(Of Decimal)).Point)
        Case X.PointType Is GetType(Date)
          val = CDbl(DirectCast(X, PlotPoint(Of Date)).Point.Ticks)
        Case X.PointType Is GetType(DateTime)
          val = CDbl(DirectCast(X, PlotPoint(Of DateTime)).Point.Ticks)
      End Select

      Return val
    End Get
  End Property

  Public ReadOnly Property YAsDouble As Double
    Get
      Dim val As Double = 0

      Select Case True
        Case Y.PointType Is GetType(Integer)
          val = CDbl(DirectCast(Y, PlotPoint(Of Integer)).Point)
        Case Y.PointType Is GetType(Double)
          val = CDbl(DirectCast(Y, PlotPoint(Of Double)).Point)
        Case Y.PointType Is GetType(Decimal)
          val = CDbl(DirectCast(Y, PlotPoint(Of Decimal)).Point)
        Case Y.PointType Is GetType(Date)
          val = CDbl(DirectCast(Y, PlotPoint(Of Date)).Point.Ticks)
        Case Y.PointType Is GetType(DateTime)
          val = CDbl(DirectCast(Y, PlotPoint(Of DateTime)).Point.Ticks)
      End Select

      Return val
    End Get
  End Property

  Public Overrides Function ToString() As String
    Return $"X {XAsDouble } Y {YAsDouble}"
  End Function

End Class

File: GenericTesting/GenericTesting/Models/Charting/PlotPoints.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;

namespace Controls.Charting
{
  public sealed class PlotPoints
  {                            
    public PlotPoint X { get; }
    public PlotPoint Y { get; }

    public PlotPoints(PlotPoint x, PlotPoint y)
    {
      this.X = x;
      this.Y = y;
      if ((Y.PointType == typeof(int)) || (Y.PointType == typeof(double)) || (Y.PointType == typeof(decimal)))
      {

      }
    }

    public double XAsDouble
    {
      get
      {
        if (X.PointType == typeof(int)) { return Convert.ToDouble(((PlotPoint<int>)X).Point); }
        else if (X.PointType == typeof(double)) { return Convert.ToDouble(((PlotPoint<double>)X).Point); }
        else if (X.PointType == typeof(decimal)) { return Convert.ToDouble(((PlotPoint<decimal>)X).Point); }
        else if (X.PointType == typeof(DateTime)) { return Convert.ToDouble(((PlotPoint<DateTime>)X).Point.Ticks); }
        else { return 0; }
      }
    }

    public double YAsDouble
    {
      get
      {
        if (Y.PointType == typeof(int)) { return Convert.ToDouble(((PlotPoint<int>)Y).Point); }
        else if (Y.PointType == typeof(double)) { return Convert.ToDouble(((PlotPoint<double>)Y).Point); }
        else if (Y.PointType == typeof(decimal)) { return Convert.ToDouble(((PlotPoint<decimal>)Y).Point); }
        else if (Y.PointType == typeof(DateTime)) { return Convert.ToDouble(((PlotPoint<DateTime>)Y).Point.Ticks); }
        else { return 0; }
      }
    }

    public override string ToString()
    {
      return $"X {XAsDouble} Y {YAsDouble}";
    } 
  }
}